packages/@aws-cdk/toolkit-lib/lib/index.ts

```diff
@@ -17,3 +17,4 @@ export * from './api/cloud-assembly';
 export * from './api/io';
 export * from './api/tags';
 export * from './api/plugin';
+export { HotswapOperationOptions } from './api/hotswap/common';
```

Comment:
Explicitly exporting only this type because there are duplicate exports if we export the entire `common` module.

```console
Module './actions' has already exported a member named 'EcsHotswapProperties'. Consider explicitly re-exporting to resolve the ambiguity.ts(2308)
Module './actions' has already exported a member named 'HotswapMode'. Consider explicitly re-exporting to resolve the ambiguity.ts(2308)
module "/Users/epolon/dev/src/github.com/aws/aws-cdk-cli/packages/@aws-cdk/toolkit-lib/lib/api/hotswap/common"

We should cleanup this up because there are currently duplicate definitions of the `HotswapMode` and `EcsHotswapProperties` class/interface. Didn't want to venture into that in this PR thought. 
```

Reply:
I tend to put the export into the action file where it's needed. This is a bit awkward in this case because of the private baseoptions, but maybe in `packages/@aws-cdk/toolkit-lib/lib/actions/deploy/index.ts` would be okay.